rxandroidble/src/test/groovy/com/polidea/rxandroidble/RxBleClientTest.groovy

```diff
@@ -79,6 +79,49 @@ class RxBleClientTest extends Specification {
         1 * bleAdapterWrapperSpy.startLeScan(_) >> true
     }
 
+    def "should start BLE scan if bluetooth is enabled before subscribing to the scan observable"() {
+        given:
+        TestSubscriber testSubscriber = new TestSubscriber<>()
+        bleAdapterWrapperSpy.hasBluetoothAdapter() >> true
+        bleAdapterWrapperSpy.isBluetoothEnabled() >> true
+
+        when:
+        def scanObservable = objectUnderTest.scanBleDevices(null)
+
+        then:
+        0 * bleAdapterWrapperSpy.isBluetoothEnabled()
+
+        when:
+        scanObservable.subscribe(testSubscriber)
+
+        then:
+        1 * bleAdapterWrapperSpy.startLeScan(_) >> true
+    }
+
+    def "should start scan for only subscribers with no exceptions from scan observable"() {
```

Comment:
Got it. Do you think this would be fine using where? 

        def "should check if all the conditions are met at the time of each subscription"() {
        given:
        def firstSubscriber = new TestSubscriber<>()
        def secondSubscriber = new TestSubscriber<>()
        bleAdapterWrapperSpy.isBluetoothEnabled() >> bluetoothEnabled >> true
        bleAdapterWrapperSpy.hasBluetoothAdapter() >> bluetoothAvailable >> true
        locationServicesStatusMock.isLocationPermissionOk() >>> [locationPermissionsOk, true]
        locationServicesStatusMock.isLocationProviderOk() >>> [locationProviderOk, true]
        def scanObservable = objectUnderTest.scanBleDevices(null)

        when:
        scanObservable.subscribe(firstSubscriber)

        then:
        firstSubscriber.assertError {
            BleScanException exception -> exception.reason == reason
        }

        when:
        scanObservable.subscribe(secondSubscriber)

        then:
        secondSubscriber.assertNoErrors()
        1 * bleAdapterWrapperSpy.startLeScan(_) >> true

        where:
        bluetoothAvailable | bluetoothEnabled | locationPermissionsOk | locationProviderOk | reason
        false              | true             | true                  | true               | BLUETOOTH_NOT_AVAILABLE
        true               | false            | true                  | true               | BLUETOOTH_DISABLED
        true               | true             | false                 | true               | LOCATION_PERMISSION_MISSING
        true               | true             | true                  | false              | LOCATION_SERVICES_DISABLED
    }